app/models/workflow_run.rb

```diff
@@ -266,6 +286,12 @@ def on_fail!
     notify!("The workflow run has failed!", :workflow_run_failed, notification_params)
   end
 
+  def on_trigger_fail!(error)
+    triggering_release.fail!
```

Comment:
This is something we cannot unfortunately enforce, but since it's making a state change on its parent release. We should drive this through `Coordinators::Signals`. Just like we have `workflow_run_finished!` we can have `workflow_run_trigger_failed!` that doesn't do much... just marks the triggering release as failed (instead of doing it inline).

Thumb-rule is: any non-internal state-change in all the "runtime" models go through coordinator, so it's easy to track to who & what is changing what.